sigmastate/src/main/scala/sigmastate/lang/Terms.scala

```diff
@@ -140,21 +140,12 @@ object Terms {
 
     protected final override def eval(env: DataEnv)(implicit E: ErgoTreeEvaluator): Any = {
       addCost(Apply.costKind)
-      if (args.isEmpty) {
-        // TODO coverage
-        val fV = func.evalTo[() => Any](env)
-        fV()
-      }
-      else if (args.length == 1) {
+      if (args.length == 1) {
```

Comment:
The change is not described in PR desc, no comments, where is the same constraint for 4.0 ?

Reply:
Already discussed, here is my answer https://github.com/ScorexFoundation/sigmastate-interpreter/pull/761#discussion_r779565072